src/lib.rs

```diff
@@ -168,18 +195,18 @@ impl Wiki {
                 }
 
                 if !path.exists() {
-                    return Ok(Response::with(status::InternalServerError));
+                    return Ok(Response::with(get_http_error_as_html(status::NotFound)));
```

Comment:
To avoid the `function(in_function(in_function(...)))` calls we ca add a new [Trait](https://doc.rust-lang.org/book/traits.html) to `iron::status::Status` which returns our HTML in case on an error.